app/src/main/java/com/babylon/wallet/android/presentation/account/composable/MetadataView.kt

```diff
@@ -36,117 +43,171 @@ import java.time.Instant
+    metadata: Metadata,
+    style: TextStyle = RadixTheme.typography.body1Regular,
+    color: Color = RadixTheme.colors.gray2,
+) {
+    MetadataKeyView(
+        modifier = modifier,
+        key = metadata.key,
+        isLocked = metadata.isLocked,
+        style = style,
+        color = color
+    )
+}
+
+@Composable
+fun MetadataKeyView(
+    modifier: Modifier = Modifier,
+    key: String,
+    isLocked: Boolean,
+    style: TextStyle = RadixTheme.typography.body1Regular,
+    color: Color = RadixTheme.colors.gray2,
 ) {
     Text(
-        modifier = modifier.padding(end = RadixTheme.dimensions.paddingMedium),
-        text = key,
-        style = RadixTheme.typography.body1Regular,
-        color = RadixTheme.colors.gray2,
-        textAlign = TextAlign.Start
+        modifier = modifier,
+        text = buildAnnotatedString {
+            append(key)
+            append(" ")
+            if (isLocked) {
+                appendInlineContent(id = "lock_icon")
+            }
+        },
+        style = style,
+        color = color,
+        textAlign = TextAlign.Start,
+        inlineContent = mapOf(
+            "lock_icon" to InlineTextContent(
+                Placeholder(style.fontSize, style.fontSize, PlaceholderVerticalAlign.TextCenter)
+            ) {
+                Icon(
+                    painter = painterResource(id = com.babylon.wallet.android.designsystem.R.drawable.ic_lock),
+                    contentDescription = null,
+                    tint = color
+                )
+            }
+        )
     )
 }
 
 @Suppress("CyclomaticComplexMethod")
 @Composable
-fun Metadata.ValueView(
+fun MetadataValueView(
     modifier: Modifier = Modifier,
-    isRenderedInNewLine: Boolean
+    metadata: Metadata,
+    isRenderedInNewLine: Boolean,
+    style: TextStyle = RadixTheme.typography.body1HighImportance,
+    color: Color = RadixTheme.colors.gray1
 ) {
     val context = LocalContext.current
-    when (this) {
+    when (metadata) {
         is Metadata.Collection, is Metadata.Map -> Text(
             modifier = modifier,
             text = stringResource(id = R.string.assetDetails_NFTDetails_complexData),
-            style = RadixTheme.typography.body1HighImportance,
-            color = RadixTheme.colors.gray1,
+            style = style,
+            color = color,
             textAlign = if (isRenderedInNewLine) TextAlign.Start else TextAlign.End,
             maxLines = 2
         )
 
-        is Metadata.Primitive -> when (valueType) {
+        is Metadata.Primitive -> when (metadata.valueType) {
             MetadataType.Bool,
             is MetadataType.Integer,
             MetadataType.Bytes,
-            MetadataType.Enum,
+            MetadataType.Enum -> Text(
+                modifier = modifier,
+                text = metadata.value,
+                style = style,
+                color = color,
+                textAlign = if (isRenderedInNewLine) TextAlign.Start else TextAlign.End,
```

Comment:
minor: maybe we can declare textAlign before `when` claus once and use it in all cases?

Reply:
This is not applicable in all cases though. There are cases where we align start only.